lib/linguist/heuristics.rb

```diff
@@ -274,7 +274,7 @@ def call(data)
     end
 
     disambiguate ".pl" do |data|
-      if /^(use v6|(my )?class|module)/.match(data)
+      if /^(use v6|(my)? class|module)/.match(data) #class might be a variable name in Perl 5
```

Comment:
> The problem is I guess in the optional my, which maybe drops the ^.

I'm not sure I follow you... What do you mean by that?

> Please check the file I've added to the test suite.

Which one is generating a false positive?